### DISCUSSION TITLE
Using USB IR Sensor (Volkszähler) with ESP8266 and Tasmota

### DISCUSSION THREAD

**Discussion by opl-git** [arendst/Tasmota#20394] [Q&A]
Hi there,
I am currently using an USB IR sensor with an Raspi Pi. It´s working fine, but it kept me wondering, if I could use the sensor with an esp8266 and Tasmota.
Here is the original source: https://wiki.volkszaehler.org/hardware/controllers/ir-schreib-lesekopf-usb-ausgang
Not much info here, so I opened the sensor and found that there is a CP2104 chip on the board.
Maybe someone with far more knowledge can help me from here. :)

Kind regards
Daniel

**Comment by sfromis** [comment]
That USB/serial adapter chip makes it harder to connect to ESP8266, without support for being an USB host. You could bypass that chip, to connect the IR signals directly to ESP8266 gpios, similar to https://wiki.volkszaehler.org/hardware/controllers/ir-schreib-lesekopf-ttl-ausgang

Many electricity meters are supported by Tasmota 
https://tasmota.github.io/docs/Smart-Meter-Interface/
